plugin/trino-hive/pom.xml

```diff
@@ -118,6 +118,64 @@
             <artifactId>units</artifactId>
         </dependency>
 
+        <dependency>
+            <groupId>com.aliyun</groupId>
+            <artifactId>datalake20200710</artifactId>
+            <version>2.0.2</version>
+            <classifier>shaded</classifier>
```

Comment:
Is it acceptable if we exclude everything in shaded datalake20200710 JAR? We will also relocate the two packages soon. The non-shaded JAR may lead to extra workload for other components in DLF.

Reply:
I checked the `shaded` classifier version of version `2.0.8` and it looks like there are additional non-relocated packages:
* `org/bouncycastle`
* `org/dom4j`
* `org/ini4j`
* `org/w3c`
* `org/xml/sax`

We need to make sure everything is shaded or use the non-shaded JAR. Another concern is that since the shaded JAR is a classifier and not the main artifact, the sources JAR does not contain sources for the relocated classes. If you think using a shaded JAR is best, can you produce a new shaded artifact that contains properly relocated sources for the relocated classes? If using `maven-shade-plugin`, you can use use this configuration:
```xml
<createSourcesJar>true</createSourcesJar>
<shadeSourcesContent>true</shadeSourcesContent>
```

However, given recent focus on the security of dependencies, we need to keep dependencies up-to-date so that they are free of known vulnerabilities / CVEs, and shaded artifacts impede our ability to do that. Thus, we would prefer not to use third party shaded artifacts when possible.

What are the concerns with using the non-shaded versions? What dependencies are required by the client code?

Reply:
It's the references in metastore-common jar that uses the shaded jar. I'll try to communicate with the author of DLF client to see if we can solve it.

> I checked the `shaded` classifier version of version `2.0.8` and it looks like there are additional non-relocated packages:
> 
> * `org/bouncycastle`
> * `org/dom4j`
> * `org/ini4j`
> * `org/w3c`
> * `org/xml/sax`
> 
> We need to make sure everything is shaded or use the non-shaded JAR. Another concern is that since the shaded JAR is a classifier and not the main artifact, the sources JAR does not contain sources for the relocated classes. If you think using a shaded JAR is best, can you produce a new shaded artifact that contains properly relocated sources for the relocated classes? If using `maven-shade-plugin`, you can use use this configuration:
> 
> ```
> <createSourcesJar>true</createSourcesJar>
> <shadeSourcesContent>true</shadeSourcesContent>
> ```
> 
> However, given recent focus on the security of dependencies, we need to keep dependencies up-to-date so that they are free of known vulnerabilities / CVEs, and shaded artifacts impede our ability to do that. Thus, we would prefer not to use third party shaded artifacts when possible.
> 
> What are the concerns with using the non-shaded versions? What dependencies are required by the client code?